src/main/java/com/openmrs/migrator/core/utilities/FileIOUtilities.java

```diff
@@ -156,4 +175,148 @@ public boolean removeDirectory(File directoryToBeDeleted) throws InvalidParamete
+    try (BufferedReader br = new BufferedReader(new FileReader(settingProperties.toFile()))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        String[] keyValue = line.split("=");
+        if ("database_name".equals(keyValue[0]) && (databaseName.equals(keyValue[1]))) {
+
+          logger.info("database name '" + databaseName + "' found in config file");
+          return Optional.of(keyValue[1]);
+        }
+      }
+      return Optional.empty();
+    }
+  }
+
+  public void setConnectionToKettleFile(String dataBaseName) throws IOException {
+    logger.info("Setting the database source  connections to the kettle.properties file");
+
+    Path kettlePath = getKettlePropertiesLocation().toPath();
+
+    Files.readAllLines(settingProperties)
+        .forEach(
+            configLine -> {
+              if (configLine.contains("username=")) {
+                username = configLine.split("=")[1];
+              }
+              if (configLine.contains("password=")) {
+                password = configLine.split("=")[1];
+              }
+              if (configLine.contains("host=")) {
+                host = configLine.split("=")[1];
+              }
+              if (configLine.contains("port=")) {
+                port = configLine.split("=")[1];
+              }
+            });
+
+    new PrintWriter(kettlePath.toFile()).close();
+
+    writeToFile(
+        kettlePath.toFile(),
+        "ETL_SOURCE_DATABASE=" + dataBaseName,
+        "ETL_DATABASE_HOST=" + host,
+        "ETL_DATABASE_PORT=" + port,
+        "ETL_DATABASE_USER=" + username,
+        "ETL_DATABASE_PASSWORD=" + password);
+  }
+
+  private File getKettlePropertiesLocation() throws IOException {
+    logger.info("Getting the default location of kettle.properties");
+    String homeDirectory = System.getProperty("user.home");
+
+    Path kettleDir = Paths.get(homeDirectory + "/" + KETTLE_DIR);
+    Path kettleFile = Paths.get(kettleDir + "/" + KETTLE_PROPERTIES);
+
+    if (!Files.exists(kettleDir)) {
+      Files.createDirectories(kettleDir);
+      Files.createFile(kettleFile);
+    }
+    if (!Files.exists(kettleFile)) {
+      Files.createFile(kettleFile);
+    }
+
+    File file = new File(homeDirectory + "/" + KETTLE_DIR + "/" + KETTLE_PROPERTIES);
+    return file;
+  }
+
+  public List<String> getAllDataBaseNamesFromConfigFile()
+      throws FileNotFoundException, IOException {
+    logger.info("retrieving all data base  names in " + settingProperties + " file");
+    List<String> names = new ArrayList<>();
+    try (BufferedReader br = new BufferedReader(new FileReader(settingProperties.toFile()))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (line.contains("database_name")) {
+          names.add(line.split("=")[1]);
+        }
+      }
+    }
+    return names;
+  }
+
+  private void writeToFile(File file, String... contents) throws IOException {
```

Comment:
this belongs here but should a public function and needs unit tests to validate (add tests for various combinations of the parameters)

Reply:
done